core/src/modules/ops/export.ts

```diff
@@ -184,22 +210,21 @@ export namespace ExportOps {
 
     if (destination) where.destinationId = destination.id;
 
-    if (saveExports) {
-      const [count] = await Export.update(
-        {
-          state: "pending",
-          sendAt: new Date(),
-          startedAt: null,
-          errorMessage: null,
-          errorLevel: null,
-          retryCount: 0,
-        },
-        { where }
-      );
+    return retryExport(where, saveExports);
+  }
 
-      return count;
-    }
+  /**
+   * Finds finds and retries a failed or canceled export by id
+   */
+  export async function retryFailedExportById(
+    exportId: string,
+    saveExports = true
+  ) {
+    const where: WhereOptions<Export> = {
+      state: { [Op.in]: ["failed", "canceled"] },
+      id: exportId,
+    };
 
-    return Export.count({ where });
+    return retryExport(where, saveExports);
```

Comment:
nitpick-ish: this saveExports stuff is really meant for previewing how many exports the bulk action would affect, which isn't really needed for the single export case. When attempting to retry a single export, if nothing has been updated I would argue that should be an error thrown back to the action.